### PR TITLE
README.md improvements (spelling, missing steps, etc.)

### PR DESCRIPTION
**Pull request by hannes-ucsc**
Just some things I ran into while working through the README.

### Test plan

Manually test the modified setup steps once.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,12 +53,12 @@ Run `source environment`  now and whenever these environment files are modified.
 * Follow the instructions in https://cloud.google.com/sdk/downloads to get the `gcloud` command line utility.
 * In the [Google Cloud Console](https://console.cloud.google.com/), select the correct Google user account on the top
   right and the correct GCP project in the drop down in the top center. Go to "IAM & Admin", then "Service accounts",
-  then click "Create service account" and select "Furnish a new private key". Create the account and download the
+  then click "Create service account" and select "Furnish a new private key". Under "Roles" select "Project – Owner", "Project – Service Account Actor" and "Cloud Functions – Cloud Function Developer". Create the account and download the
```
Comment (hannes-ucsc) [inline README.md]: My hunch is that this is too permissive and, I think, should be scoped to the project via IAM but I don't know GCP well enough.

**Comment by hannes-ucsc** [comment]
Am I correct that the creation of the API gateway is missing from the README?

**Comment by hannes-ucsc** [comment]
Nope, it's there. Ignore the previous question.

File: README.md [hunk 2]
```diff
@@ -42,83 +34,109 @@ Run `source environment`  now and whenever these environment files are modified.
...
 
 ##### AWS
-* Follow the instructions in http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html to get the
-  `aws` command line utility.
-* Create an S3 bucket that you want DSS to use.
-* In `environment.local`, set the environment variable `DSS_S3_BUCKET_TEST`.
-* If you wish to run the unit tests, you must create a second S3 bucket to store the test fixtures, and set the
-  environment variable `DSS_S3_BUCKET_TEST_FIXTURES` to the name of that bucket.
+
+* Follow the instructions in http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html to get the `aws` command line utility.
```
Comment (ttung) [inline README.md]: There is inconsistent usage of bulleted lists vs numbered lists.  May not be relevant to your change, but FYI.

File: README.md [hunk 3]
```diff
@@ -42,83 +34,109 @@ Run `source environment`  now and whenever these environment files are modified.
...
+
+* Create an S3 bucket that you want DSS to use and in `environment.local`, set the environment variable `DSS_S3_BUCKET` to the name of that bucket. Make sure the bucket region is consistent with `AWS_DEFAULT_REGION` in `environment.local`.
```
Comment (ttung) [inline README.md]: I don't know if it's confusing that we're telling people to set environment variables vs `environment.local`.

File: README.md [hunk 4]
```diff
@@ -167,18 +181,12 @@ indexed metadata.
...
 
 #### Authorizing Travis CI to deploy
-Encrypted environment variables give Travis CI the AWS credentials needed to run the tests and deploy the app. Run
-`scripts/authorize_aws_deploy.sh IAM-PRINCIPAL-TYPE IAM-PRINCIPAL-NAME` (e.g. `authorize_aws_deploy.sh user hca-test`)
-to give that principal the permissions needed to deploy the app. Because this is a limited set of permissions, it does
-not have write access to IAM. To set up the IAM policies for resources in your account that the app will use, run `make
-deploy` using privileged account credentials once from your workstation. After this is done, Travis CI will be able to
-deploy on its own. You must repeat the `make deploy` step from a privileged account any time you change the IAM policies
-in `policy.json.template` files.
+
```
Comment (ttung) [inline README.md]: any time you change the IAM policies templates in `iam/policy-templates/`

File: README.md (hunk 2 above)
Comment (hannes-ucsc) [reply README.md]: I'll fix it. a555277507219b2d6357ae59bf2628e264045fab makes that a lot easier.

File: README.md (hunk 3 above)
Comment (hannes-ucsc) [reply README.md]: It was clear to me when I went through it. It's a common pattern.

File: README.md (hunk 4 above)
Comment (hannes-ucsc) [reply README.md]: Fixed.

File: README.md [hunk 5]
```diff
@@ -42,83 +34,110 @@ Run `source environment`  now and whenever these environment files are modified.
...
+8.  Generate OAuth application secrets to be used for your instance: 
+
+    1) Go to https://console.developers.google.com/apis/credentials (you may have to select Organization and Project again)
+    
+	2) Click *Create Credentials* and select *OAuth client*
+    
+	3) For *Application type* choose *Other*  
+    
+	4) Under application name, use `hca-dss-` followed by the stage name, i.e. the value of DSS_DEPLOYMENT_STAGE. This is a convention only and carries no technical significance.
```
Comment (ttung) [inline README.md]: `i.e.,`

File: README.md [hunk 6]
```diff
@@ -42,83 +34,110 @@ Run `source environment`  now and whenever these environment files are modified.
...
+
+Note: The certificate should be in the same region as the API gateway or, if that's not possible, in `us-east-1`. If the ACM certificate's region is `us-east-1` and the API gateway is in another region, the type of the custom domain name must be *Edge Optimized*. Provisioning such a domain name typically takes up to 40 minutes because the certificate needs to be replicated to all involved CloudFront edge servers. The corresponding record set in Route 53 needs to be an **alias** A record, not a CNAME or a regular A record, and it must point to the CloudFront host name associated with the edge-optimized domain name. Starting November 2017, API gateway supports regional certificates, i.e. certificates in regions other than `us-east-1`. This makes it possible to match the certificate's region with that of the API gateway. and cuts the provisioning of the custom domain name down to seconds. Simply create the certificate in the same region as that of the API gateway, create a custom domain name of type *Regional* and in Route53 add a CNAME recordset that points to the gateway's canonical host name.
```
Comment (ttung) [inline README.md]: i.e.,

File: README.md [hunk 7]
```diff
@@ -42,83 +34,110 @@ Run `source environment`  now and whenever these environment files are modified.
...
+9.  Create a Google Cloud Storage bucket and in `environment.local`, set the environment variable `DSS_GS_BUCKET` to the name of that bucket. Make sure the bucket region is consistent with `GCP_DEFAULT_REGION` in `environment.local`.
+
+10. If you wish to run the unit tests, you must create two more buckets, one for test data and another for test fixtures, and set the environment variables `DSS_GS_BUCKET_TEST` and `DSS_GS_BUCKET_TEST_FIXTURES` to the names of those buckets.
+
+Hint: To create GCS buckets from the command line, use `gsutil mb -c regional -l REGION gs://BUCKET_NAME/`.
 
 ##### Azure
-* Set the environment variables `AZURE_STORAGE_ACCOUNT_NAME` and `AZURE_STORAGE_ACCOUNT_KEY`.
+
+1. Set the environment variables `AZURE_STORAGE_ACCOUNT_NAME` and `AZURE_STORAGE_ACCOUNT_KEY`.
```
Comment (mikebaumann) [inline README.md]: Why even include an Azure section in the readme at this point?
I don't believe we have any support for it at all, currently.

**Pull request merged**